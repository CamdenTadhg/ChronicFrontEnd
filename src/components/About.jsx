import React from 'react';
import {Row, Container} from 'reactstrap';
import javascript from '../assets/javascript.webp';
import node from '../assets/node_logo.png';
import postgresql from '../assets/PostgreSQL-Logo.jpg';
import vite from '../assets/vite.jpg';
import react from '../assets/react.jpg';
import redux from '../assets/redux.jpg';

function About() {
    return (
        <Container fluid className='p-0 white-page'>
            <Row className='navbar-space m-0 p-0'>
            </Row>
            <Row className='about-content m-0 p-0'>
                <h2 className='page-title'>ABOUT CHRONIC</h2>
                <div className='text-content'>
                    <p>Take charge of your health with Chronic. No more digging through notes or trying to remember what you felt like last week. Everything you need is in one place -- your symptoms, your meds, and the trends that tell your story.</p>
                    <p>Spot patterns in your health that you might not have noticed before with simple graphs. Is a new treatment working? Are your symptoms getting worse? You'll see it clearly. When it's time to see your doctor, share your up-to-date data with confidence. Better communication means better care and we've got you covered. </p>
                    <p>The more you track, the more control you'll have. It's your health, organized, insightful, and ready for whatever comes next. </p>
                    <p className='second-section'>This website was built with:</p>
                    <div className='logos'>
                        <img className="tech-logo-tall" src={javascript}/>
                        <a href="https://nodejs.org/en"><img className="tech-logo-wide" src={node}/></a>
                        <a href="https://www.postgresql.org/"><img className="tech-logo-wide" src={postgresql}/></a>
                        <a href="https://vite.dev/"><img className="tech-logo-tall" src={vite}/></a>
                        <a href="https://react.dev/"><img className="tech-logo-tall" src={react}/></a>
                        <a href="https://redux.js.org/"><img className="tech-logo-tall" src={redux}/></a>
                    </div>
                </div>
            </Row>
        </Container>
    )
}

export default About;