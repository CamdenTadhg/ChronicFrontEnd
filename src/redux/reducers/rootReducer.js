import {combineReducers} from 'redux';
import profileReducer from './profileReducer';
import trackingReducer from './trackingReducer';
import dataReducer from './dataReducer';
import latestReducer from './latestReducer';

const rootReducer = combineReducers({
    profile: profileReducer,
    tracking: trackingReducer,
    data: dataReducer,
    latest: latestReducer
});

export default rootReducer;