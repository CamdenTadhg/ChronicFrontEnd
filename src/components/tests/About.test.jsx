import React from 'react';
import {render} from '@testing-library/react';
import {it, expect} from 'vitest';
import '@testing-library/jest-dom';
import {MemoryRouter} from 'react-router-dom';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import About from '../About';
import rootReducer from '../../redux/reducers/rootReducer';

it('renders without errors', () => {
    const store = configureStore({reducer: rootReducer});
    render (
        <Provider store={store}>
            <MemoryRouter>
                <About/>
            </MemoryRouter>
        </Provider>
    );
});

it('matches the snapshot', () => {
    const store = configureStore({reducer: rootReducer});
    const about = render (
        <Provider store={store}>
            <MemoryRouter>
                <About/>
            </MemoryRouter>
        </Provider>
    );
    expect(about).toMatchSnapshot();
});

it('displays the appropriate content', () => {
    const store = configureStore({reducer: rootReducer});
    const {getByText} = render (
        <Provider store={store}>
            <MemoryRouter>
                <About />
            </MemoryRouter>
        </Provider>
    );
    expect(getByText('ABOUT CHRONIC')).toBeInTheDocument();
})