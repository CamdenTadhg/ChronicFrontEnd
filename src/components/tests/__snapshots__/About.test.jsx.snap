// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-0 white-page container-fluid"
      >
        <div
          class="navbar-space m-0 p-0 row"
        />
        <div
          class="about-content m-0 p-0 row"
        >
          <h2
            class="page-title"
          >
            ABOUT CHRONIC
          </h2>
          <div
            class="text-content"
          >
            <p>
              Take change of your health with Chronic. No more digging through notes or trying to remember what you felt like last week. Everything you need is in one place -- your symptoms, your meds, and the trends that tell your story.
            </p>
            <p>
              Spot patterns in your health that you might not have noticed before with simple graphs. Is a new treatment working? Are your symptoms getting worse? You'll see it clearly. When it's time to see your doctor, share your up-to-date data with confidence. Better communication means better care and we've got you covered. 
            </p>
            <p>
              The more you track, the more control you'll have. It's your health, organized, insightful, and ready for whatever comes next. 
            </p>
            <p
              class="second-section"
            >
              This website was built with:
            </p>
            <div
              class="logos"
            >
              <img
                class="tech-logo-tall"
                src="/src/assets/javascript.webp"
              />
              <a
                href="https://nodejs.org/en"
              >
                <img
                  class="tech-logo-wide"
                  src="/src/assets/node_logo.png"
                />
              </a>
              <a
                href="https://www.postgresql.org/"
              >
                <img
                  class="tech-logo-wide"
                  src="/src/assets/PostgreSQL-Logo.jpg"
                />
              </a>
              <a
                href="https://vite.dev/"
              >
                <img
                  class="tech-logo-tall"
                  src="/src/assets/vite.jpg"
                />
              </a>
              <a
                href="https://react.dev/"
              >
                <img
                  class="tech-logo-tall"
                  src="/src/assets/react.jpg"
                />
              </a>
              <a
                href="https://redux.js.org/"
              >
                <img
                  class="tech-logo-tall"
                  src="/src/assets/redux.jpg"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-0 white-page container-fluid"
    >
      <div
        class="navbar-space m-0 p-0 row"
      />
      <div
        class="about-content m-0 p-0 row"
      >
        <h2
          class="page-title"
        >
          ABOUT CHRONIC
        </h2>
        <div
          class="text-content"
        >
          <p>
            Take change of your health with Chronic. No more digging through notes or trying to remember what you felt like last week. Everything you need is in one place -- your symptoms, your meds, and the trends that tell your story.
          </p>
          <p>
            Spot patterns in your health that you might not have noticed before with simple graphs. Is a new treatment working? Are your symptoms getting worse? You'll see it clearly. When it's time to see your doctor, share your up-to-date data with confidence. Better communication means better care and we've got you covered. 
          </p>
          <p>
            The more you track, the more control you'll have. It's your health, organized, insightful, and ready for whatever comes next. 
          </p>
          <p
            class="second-section"
          >
            This website was built with:
          </p>
          <div
            class="logos"
          >
            <img
              class="tech-logo-tall"
              src="/src/assets/javascript.webp"
            />
            <a
              href="https://nodejs.org/en"
            >
              <img
                class="tech-logo-wide"
                src="/src/assets/node_logo.png"
              />
            </a>
            <a
              href="https://www.postgresql.org/"
            >
              <img
                class="tech-logo-wide"
                src="/src/assets/PostgreSQL-Logo.jpg"
              />
            </a>
            <a
              href="https://vite.dev/"
            >
              <img
                class="tech-logo-tall"
                src="/src/assets/vite.jpg"
              />
            </a>
            <a
              href="https://react.dev/"
            >
              <img
                class="tech-logo-tall"
                src="/src/assets/react.jpg"
              />
            </a>
            <a
              href="https://redux.js.org/"
            >
              <img
                class="tech-logo-tall"
                src="/src/assets/redux.jpg"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
